app/Database/CourseQueries.hs

```diff
@@ -271,3 +273,22 @@ queryGraphs =
         do graphs :: [Entity Graph] <- selectList [] []
            return $ createJSONResponse graphs
 
+-- ========================================================
+
+getLectureTime :: [String] -> IO (Maybe [Time])
+getLectureTime [code, section, session] = runSqlite databasePath $ do 
```

Comment:
Also keep the return type as `[Time]` by returning an empty list if you get `Nothing`.